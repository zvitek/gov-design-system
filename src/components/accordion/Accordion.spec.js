import { shallowMount } from '@vue/test-utils'
import GovAccordion from '@components/accordion/Accordion'

let wrapper

describe('GovAccordion', () => {
    describe('default', () => {
        beforeEach(() => {
            wrapper = shallowMount(GovAccordion)
        })

        it('is called', () => {
            expect(wrapper.name()).toBe('GovAccordion')
            expect(wrapper.isVueInstance()).toBeTruthy()
        })

        it('render correctly', () => {
            expect(wrapper.html()).toMatchSnapshot()
        })
    })
    it('should have default slot', () => {
        const slotDefault = '<div> Content </div>'
        wrapper = shallowMount(GovAccordion, {
            slots: {
                default: slotDefault
            }
        })
        expect(wrapper.find('.gov-accordion :first-child').html()).toBe(slotDefault)
    })
    it('should be small', () => {
        wrapper.setProps({
            small: true
        })
        expect(wrapper.classes()).toContain('gov-accordion--small')
    })

    it('should be large', () => {
        wrapper.setProps({
            large: true
        })
        expect(wrapper.classes()).toContain('gov-accordion--large')
    })

    it('should be without inner offset', () => {
        wrapper.setProps({
            noInnerOffset: true
        })
        expect(wrapper.classes()).toContain('gov-accordion--no-inner-offset')
    })
})
