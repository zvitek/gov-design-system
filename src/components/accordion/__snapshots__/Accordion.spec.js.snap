// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GovAccordion default render correctly 1`] = `<div class="gov-accordion"></div>`;
